### ISSUE TITLE
Analyzer fix for handling invalid XML in seealso href attributes

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#54902]
**Brief description:**

I have this:
````cs
/// <summary>
/// Foo
/// </summary>
/// <seealso href="https://www.example.com/foo?q=bar&qux=2"/>
````

That gives:
> XML comment has badly formed XML -- 'Reference to undefined entity 'qux'.' csharp(CS1570)

The problem is the `&` is invalid XML.

Workarounds:
- convert `&` to `&amp;`
- disable the warning with a `#pragma warning disable CS1570`
- `<NoWarn>1570</NoWarn>`
- find (using regex) and replace `&` with `&amp;`

These are non ideal and a waste of time - I just want to paste my URLs into the code and move on.

Could you please consider:
1. Allowing us to configure the analyzer to allow ampersands
2. OR BETTER, have an analyzer fix to automatically fix the address as it's pasted it in a `seealso href` attribute value (like we already have when pasting in a json string -- it doubles `\` characters automatically to keep paths valid)

**Languages applicable:**
C#

**Code example that the analyzer should report:**
As above.

**Additional information:**
Originally posted at [StackOverflow](https://stackoverflow.com/q/68324710/9971404).

**Documentation requirements:**
When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by Youssef1313** [comment]
Similar to https://github.com/dotnet/roslyn/issues/54418 (they could be made as the same codefix).

**Comment by jinujoseph** [comment]
cc @sharwell
